src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
```

Comment:
@mzgol again, thank you for the follow up

> This would be going even further; we'd get automatically defined all ES6 globals & syntax which not all our supported browsers implement.

Just to clarify, are you referring to jQuery 3 or all versions?

> On the compat branch we even explicitly define es3: true so that jsHint won't e.g. allow us to introduce trailing commas in array/object literals (yes, it did happen).
> I think turning esnext on could be dangerous here.

Yeah, this is very compelling and I hadn't even considered it when I first asked—thanks for the reality check :D